lib/cdo/db.rb

```diff
@@ -14,6 +15,13 @@ def sequel_connect(writer, reader)
 
   db.extension :server_block
 
+  # Check if validation frequency is valid and then use it to specify how often to
+  # verify the db connection
+  if !validation_frequency.nil? && (validation_frequency == -1 || validation_frequency > 0)
```

Comment:
nit: why double-negative (`! variable.nil?`). Why not simply `if validation_frequency && ...`?

or, with try:
```ruby
if validation_frequency.try {|f| f == -1 || f > 0}
```